packages/core/src/utils/resolveComponent.ts

```diff
@@ -1,34 +1,20 @@
 import { ERROR_NOT_IN_RESOLVER } from '@craftjs/utils';
+import React from 'react';
 import invariant from 'tiny-invariant';
 
 import { Resolver } from '../interfaces';
 
+let currResolver: Resolver | undefined = undefined;
+const reversedResolver = new Map<React.ElementType | string, string>();
+
 export const resolveComponent = (
   resolver: Resolver,
   comp: React.ElementType | string
 ) => {
   const componentName = (comp as any).name || (comp as any).displayName;
-
-  const getNameInResolver = () => {
-    if (resolver[componentName]) {
-      return componentName;
-    }
-
-    for (let i = 0; i < Object.keys(resolver).length; i++) {
-      const name = Object.keys(resolver)[i];
-      const fn = resolver[name];
-
-      if (fn === comp) {
-        return name;
-      }
-    }
-
-    if (typeof comp === 'string') {
-      return comp;
-    }
-  };
-
-  const resolvedName = getNameInResolver();
+  const resolvedName = resolver[componentName]
+    ? componentName
+    : searchComponentInResolver(resolver, comp);
```

Comment:
nit: Think we can just simplify this to `searchComponentInResolver` in all cases since it won't really impact efficiency after the initial call:

```tsx
const resolvedName = searchComponentInResolver(resolver, comp);
```